### ISSUE TITLE
Blazor WebAssembly 3.2.0-preview3.20168.3, code behind does not work when @page directive has a parameter

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#20727]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/984782/blazor-webassembly-320-preview3201683-code-behind.html)._

---
I am migrating code that works in Blazor Server App on Visual Studio 16.5.2.

The project has 7 Components with code behind classes all but one works,.

Document.razor component:-
@page "/documents/{key}"
@inherits DocumentBase
    <div  code removed for brevity...
    </div> 

The Code behind:-
public class DocumentBase : ComponentBase
{
   [Parameter]
   public string Key { get; set; }
   public string Text { get; set; }
   protected override async Task OnInitializedAsync()
   {
      Text = GetDocument(Key);  //Code removed for brevity...
   }
}

When making the following call;
 NavManager.NavigateTo($"/documents/{Document.BlobName}");

I get the following error:-
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
blazor.webassembly.js (1,36066)

Unhandled exception rendering component: Cannot provide a value for property 'Key' on type 'RhzHome.Client.Pages.Document'. There is no registered service of type 'System.String'.

If I move the Code behind to @code{ }
It works as expected.

---
#### Visual Studio Feedback System on 4/10/2020, 01:19 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by mkArtakMSFT** [comment]
Thank you for filing this issue. In order for us to investigate this issue, please provide a minimalistic repro project (ideally a GitHub repo) that illustrates the problem.

**Comment by richardhz** [comment]
Totally embarrassed! 
While migrating I commented out some code, so the  code behind was actually 
public class DocumentBase : ComponentBase
{
   [Inject]
   //protected ICacheService CacheService { get; set; }
   [Parameter]
   public string Key { get; set; }
   public string Text { get; set; }
   protected override async Task OnInitializedAsync()
   {
      Text = GetDocument(Key); //Code removed for brevity...
   }
}

Removing the Inject attribute removes the problem. Sorry for any time wasted. 
I guess I got so used to Visual Studio assistance the I didn't read code as carefully as I should.
